index.js

```diff
@@ -13,133 +13,167 @@
  * babel cli:
  *
  * babel [your-source-dir] --plugins=glamorous-to-emotion --presets=react,etc... --out-dir=[your-source-dir]
- * 
+ *
  * A demo can be seen at:
  * https://astexplorer.net/#/gist/7bc4771564a12c9f93c4904b3934aa1c/latests
  */
 
-module.exports = function (babel) {
-    const { types: t } = babel;
+module.exports = function(babel) {
+  const {types: t} = babel;
+
+  const fixContentProp = args => {
+    return args.map(argument => {
+      if (argument.type !== "ObjectExpression") {
+        return argument;
+      }
+      return {
+        ...argument,
+        properties: argument.properties.map(prop => {
+          if (prop.key.name !== "content") {
+            return prop;
+          }
 
-    return {
-        visitor: {
-            /**
-             * Find all JSX elements that use the old
-             * <glamorous.Something> notation and rewrite them to be
-             * plain ole somethings.
-             */
-            JSXElement(path) {
-                path.traverse({
-                    "JSXOpeningElement|JSXClosingElement"(path) {
-                        if (!t.isJSXMemberExpression(path.node.name)) {
-                            return;
-                        }
-                        if (path.node.name.object.name !== "glamorous") {
-                            return;
-                        }
-                        path.node.name = t.identifier(
-                            path.node.name.property.name.toLowerCase()
-                        );
-                    }
-                });
-            },
+          // Add quotes to the content property.
+          return {
+            ...prop,
+            value: t.stringLiteral(`"${prop.value.value}"`),
+          };
+        }),
+      };
+    });
+  };
 
-            /**
-             * Find glamorous import statements, including ThemeProvider
-             * imports and rewrite them to be emotion imports.
-             */
-            ImportDeclaration(path, { opts }) {
-                if (path.node.source.value !== "glamorous") {
-                    return;
-                }
+  const transformJSXAttributes = attrs => {
+    if (!attrs) return [];
+    const appendToCss = [];
+    let cssAttr = null;
+    const newAttrs = attrs.filter(attr => {
+      const {value, name, type} = attr;
+      if (type === "JSXSpreadAttribute") return true;
+      if (name.name === "css") {
+        cssAttr = attr;
+      } else {
+        // event handlers are no CSS Props!
+        if (name.name.indexOf("on") === 0) return true;
 
-                // First, check for ThemeProvider and update that.
-                const themeProviderIndex = path.node.specifiers.findIndex(
-                    specifier => specifier.local.name === "ThemeProvider"
-                );
+        appendToCss.push({
+          name: t.identifier(name.name),
+          value: value.type === "JSXExpressionContainer" ? value.expression : value,
```

Comment:
Would the `t.isJSXExpressionContainer` function work for this validation instead? I think it's more future proof because it could handle more imperative validations if needed in the future.

Reply:
This probably applies to `type === "XXX"` checks, of which I did quite a lot.